packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -73,6 +76,10 @@ function HeaderToolbar() {
 	/* translators: accessibility text for the editor toolbar */
 	const toolbarAriaLabel = __( 'Document tools' );
 
+	const toggleListView = useCallback(
+		() => setIsListViewOpened( ( isOpen ) => ! isOpen ),
```

Comment:
We can use a functional update is avoid a closure with a stale value, or needing to specify state as a dependency. 

https://reactjs.org/docs/hooks-reference.html#functional-updates